li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -372,17 +458,44 @@ public Long safeOffset(TopicPartition tp) {
 
   @Override
   public void close() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    close(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
   }
 
   @Override
   public void close(long timeout, TimeUnit timeUnit) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (_consumers.isEmpty()) {
+      LOG.warn("No consumers to close for cluster group {}", _clusterGroup);
+      return;
+    }
+
+    LOG.info("Closing LiKafkaConsumer for cluster group {} in {} {}...", _clusterGroup, timeout, timeUnit);
+
+    long startTimeMs = System.currentTimeMillis();
+    long deadlineTimeMs = startTimeMs + timeUnit.toMillis(timeout);
+    CountDownLatch countDownLatch = new CountDownLatch(_consumers.entrySet().size());
+    for (LiKafkaConsumer<K, V> consumer : _consumers.values()) {
+      new Thread(() -> {
```

Comment:
make thread a daemon, give it a name, install an uncaught ExceptionHandler

Reply:
done